experiments/customui/parent.js

```diff
@@ -373,12 +575,31 @@ var ex_customui = class extends ExtensionCommon.ExtensionAPI {
           splitter.width = width; // do not add px
           container.appendChild(splitter);
 
-          let parentGridTemplate = window.getComputedStyle(container).gridTemplate;
           container.style.setProperty(`--${splitter.id}-width`,width+"px");
-          //TODO generate this edit: add 2 columns
-          container.style.gridTemplate = `"folders folderPaneSplitter threads ${splitter.style.gridArea} ${sidebar.style.gridArea}" minmax(auto, 1fr) "folders folderPaneSplitter messagePaneSplitter ${splitter.style.gridArea} ${sidebar.style.gridArea}" min-content "folders folderPaneSplitter message ${splitter.style.gridArea} ${sidebar.style.gridArea}" minmax(auto, var(--messagePaneSplitter-height)) / minmax(auto, var(--folderPaneSplitter-width)) min-content minmax(auto, 1fr) min-content minmax(auto, var(--${splitter.id}-width))`;
+
+          // set an observer on the container to act on class changes
+          const updateGridTemplate = (mutationList, observer) => {
+            addColumnsToElementGridTemplate(
+              container,
+              window,
+              [
+                {
+                  areaname:splitter.style.gridArea,
+                  columnTemplate:"min-content/*customuisplitter*/",
+                },
+                {
+                  areaname:sidebar.style.gridArea,
+                  columnTemplate:`minmax(auto, var(--${splitter.id}-width))`
+                }
+              ]
+            );
+          };
+          updateGridTemplate();
+          const observer = new window.MutationObserver(updateGridTemplate);
+          window.customuiObserver = observer;
```

Comment:
This looks dangerous in situations in which you have multiple sidebars (possibly in different add-ons using this experiment), as you could overwrite the observer set for the previous sidebar.

Reply:
In fact, it's ok they share the same observer (for now) as it essentially does the same and in some cases, we only want it to run only once. Multiple sidebars are currently vertically stacked without issue, which may be desired behavior.
If there are more add-ons using this, it could probably be looked after to have optional vertical or horizontal stacking.

Reply:
At the very least you'd leak the observer (as you don't disconnect the previous one). Either way, just prefix the variable with `context.contextId` and be done with it?